core/src/test/scala/org/apache/spark/scheduler/ReplayListenerSuite.scala

```diff
@@ -151,7 +153,10 @@ class ReplayListenerSuite extends SparkFunSuite with BeforeAndAfter with LocalSp
    * assumption that the event logging behavior is correct (tested in a separate suite).
    */
   private def testApplicationReplay(codecName: Option[String] = None) {
-    val logDirPath = Utils.getFilePath(testDir, "test-replay")
+    val logDir = new File(testDir.getAbsolutePath, "test-replay")
+    // Here, it creates `Path` from the URI instead of the absolute path for the explicit file
+    // scheme so that the string representation of this `Path` has leading file scheme correctly.
+    val logDirPath = new Path(logDir.toURI)
```

Comment:
This test itself was added long time ago but looks there was a recent change related with this code path - https://github.com/apache/spark/commit/edcb878e2fbd0d85bf70614fed37f4cbf0caa95e.

https://github.com/apache/spark/blob/90195485a6e8b95b1cb4c5606e508a05b6fdba11/core/src/test/scala/org/apache/spark/scheduler/ReplayListenerSuite.scala#L179